src/platform/site-wide/banners/components/MaintenanceBanner/index.js

```diff
@@ -0,0 +1,108 @@
+// Node modules.
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import AlertBox from '@department-of-veterans-affairs/formation-react/AlertBox';
+import classnames from 'classnames';
+import moment from 'moment';
+// Relative imports.
+import environment from 'platform/utilities/environment';
+import localStorage from 'platform/utilities/storage/localStorage';
+import config from './config';
+
+export const MAINTENANCE_BANNER = 'MAINTENANCE_BANNER';
+
+export class MaintenanceBanner extends Component {
+  static propTypes = {
+    content: PropTypes.string.isRequired,
+    expiresAt: PropTypes.object.isRequired,
+    id: PropTypes.string.isRequired,
+    startsAt: PropTypes.object.isRequired,
+    title: PropTypes.string.isRequired,
+    warnContent: PropTypes.string,
```

Comment:
I think we were asked to add some enforcement for the 300-length limit on the content so maybe we could throw an explicit error if the text length exceeds that. That could make sense as a follow-up PR though and to ahve this lay the foundation

Reply:
Oh you're right! As the timestamp changes that length will change, too 🤔 Hmm...

Reply:
Yeah, and the length also doesn't include HTML tags (only the text content itself) so it's not actually a super easy problem

Reply:
Ooo, do you think I should set inner html here? I wasn't planning on the maintenance banner having jsx elements in it other than just a plain string with formatted startsAt, etc. Currently it just shows this non-html string.

Reply:
Good question. I'm a little torn because on one hand we want to standardize the messaging but we also want to cap the message at a 300-character limit, which suggests we do intend to customize the message every now and then. How you have it now as a plain string seems totally fine as a starting point